go/vt/vtgate/planner.go

```diff
@@ -43,10 +52,53 @@ func NewPlanner(vschema *vindexes.VSchema, cacheSize int) *Planner {
 	return plr
 }
 
+// LoadVSchema loads the VSchema from the topo. The function does
+// not return an error. It instead logs warnings on failure.
+func (plr *Planner) LoadVSchema() {
+	formal := &vindexes.VSchemaFormal{
+		Keyspaces: make(map[string]vindexes.KeyspaceFormal),
+	}
+	keyspaces, err := plr.serv.GetSrvKeyspaceNames(context.TODO(), plr.cell)
```

Comment:
Yeah. The watcher should be per-keyspace. But the loader will need to re-read all vschemas to recompute dups. We could cache the vschemas, but it's not worth it.